src/ert/ensemble_evaluator/dispatch.py

```diff
@@ -39,17 +40,21 @@ def _done_callback(self, *args):
 
     async def _work(self):
         t0 = time.time()
-        event_buffer, self._buffer = (
-            self._buffer[: self._max_batch],
-            self._buffer[self._max_batch :],
-        )
+        event_buffer = []
+        for _ in range(self._max_batch):
```

Comment:
There is a slicing analogy / substitution for deque: https://stackoverflow.com/questions/7064289/use-slice-notation-with-collections-deque

Reply:
tried to do a bit of benchmarking, not sure how reliable the data i produced was, but i tested the same thing with lists and dequeu, which is supposed to resemble what happens in ert, and the list approach was 2 times faster. so i'm dropping this commit altogether.

Reply:
just for reference, i was a bit discouraged of trying to slice the deque using itertools by this quote that i found in [a forum](https://discuss.python.org/t/slicing-notation-with-deque-and-multi-dimensional-slicing-on-data-structures/2791/3):

> The deque type is a linked list and indexing past 0 and -1 is inefficient (read, takes O(N) time), so slicing those doesn’t make much sense.

it also raises the good (or at least practically, it should be solved) question of what's the return type of the call to islice (it's not a deque a priori), and the islice call doesn't pop elements, so we need to calls, and ... bla bla bla this was just to share some more thoughts around this idea